File: ii-service/src/main/java/com/manymonkeys/service/auth/UserService.java
package com.manymonkeys.service.auth;

import com.manymonkeys.model.user.User;

/**
 * @author Ilya Pimenov
 *         Xaton Proprietary
 */
public interface UserService {

    User createUser(User user);

    User getUser(String login);

    User setPassword(User user, String password);

    Boolean checkPassword(User user, String password);
}

File: ii-service/src/main/scala/com/manymonkeys/research/service/auth/UserService.scala
package com.manymonkeys.research.service.auth

import com.manymonkeys.model.auth.User
import com.manymonkeys.model.cinema.Movie

/**
 * @author Ilya Pimenov
 *         Owl Proprietary
 */
trait UserService {

  def createUser(user: User): User

  def getUser(login: String): User

  def setPassword(user: User, password: String): User

  def checkPassword(user: User, password: String): Boolean

  def follow(follower: User,  followed: User): User

  def unfollow(follower: User,  followed: User): User

  def like(user: User, movie: Movie): User

  def rate(user: User,  movie: Movie, rating: Double): User

}

